numpy/lib/format.py

```diff
@@ -177,6 +177,8 @@
 MAGIC_LEN = len(MAGIC_PREFIX) + 2
 ARRAY_ALIGN = 64 # plausible values are powers of 2 between 16 and 4096
 BUFFER_SIZE = 2**18  # size of buffer for reading npz files in bytes
+# allow growth within the address space of a 64 bit machine along one axis
+GROWTH_AXIS_MAX_DIGITS = 20  # = len(str(2**64-1))
```

Comment:
We actually only support `2**63-1` (using signed integers internally).  But this is still fine...

Reply:
As I've meant, `size_t` is unsigned, so even if Python and/or Numpy has signed integers internally, with some other program one might theoretically use that remaining bit.
Actually, I had some extra thoughts about this: what if Numpy (or some other language using the .npy format), today or at some point in the future, supports/will support sub-byte datatypes like `int4`, `int2`, or `int1`? In Deep Learning, `int4` already is a thing. Let's construct something like `arr = np.array(range(17), dtype='int1')`. Then, since pointers still operate bytewise, `arr[15]=arr[1*8+7]` would be some 'resolve byte pointer and mask the return value'-operation (which could, depending on the arithmetic done on the array, be accelerated by the compiler and replaced by some higher level instructions, adding two `int1` is basically just an xor). However, the index could then become larger than `2**64-1`, actually up to `8*2**64-1`. I mean, does not make a big difference, but I'll give it 21 bytes with the next push.

Reply:
All very hypothetical :).  I honestly don't care what you use.  I just thought I would note it since NumPy signed is sufficient.  As you said elsewhere: even 2**63 bits is absurdly much anyway...